core/src/consensus.rs

```diff
@@ -921,11 +921,12 @@ impl Tower {
             }
 
             if let Some(checked_slot) = checked_slot {
-                // This is really special, only if tower is initialized (root = slot 0) for genesis and contains
-                // a vote (= slot 0) for the genesis, the slot 0 can repeat only once
-                let voting_from_genesis = *slot_in_tower == checked_slot && *slot_in_tower == 0;
+                // This is really special, only if tower is initialized and contains
+                // a vote for the root, the root slot can repeat only once
+                let voting_for_root =
+                    *slot_in_tower == checked_slot && *slot_in_tower == tower_root;
 
-                if !voting_from_genesis {
```

Comment:
Considering this false panic, it's so tempting to just give up this asserting entirely and call `self.initialize_lockouts(|v| v.slot > tower_root);` and get done by forcibly erasing any odd state. But, I rather want to keep this assertion by tweaking like this.

This is to catch a really ill-state of badly ordered vote account / tower if it ever happens in the future.